src/apps/chifra/cmd/abis/handle_find.go

```diff
@@ -13,12 +13,106 @@
 package abis
 
 import (
+	"bufio"
+	"bytes"
+	"encoding/hex"
 	"fmt"
+	"log"
+	"os"
+	"runtime"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	ants "github.com/panjf2000/ants/v2"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/config"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/output"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/utils"
 )
 
 // HandleFind loads manifest, sorts pins and prints them out
-func HandleFind(terms []string) {
-	for i, term := range terms {
-		fmt.Println("term[", i, "]: ", term)
+func HandleFind(args []string) {
+	testMode := utils.IsTestMode()
+	apiMode := utils.IsApiMode()
+	if apiMode || output.Format == "" || output.Format == "none" {
+		output.Format = "json"
+	}
+
+	var arguments [][]byte
+	for _, arg := range args {
+		if arg[:2] == "0x" {
+			asBytes, err := hex.DecodeString(arg[2:])
+			if err != nil {
+				fmt.Printf("Couldn't parse argument %s\n", arg)
+			} else {
+				arguments = append(arguments, asBytes)
+			}
+		}
+	}
+
+	var wg sync.WaitGroup
+	checkOne, _ := ants.NewPoolWithFunc(runtime.NumCPU()*2/3, func(input interface{}) {
+		defer wg.Done()
+		str := input.(string)
+		byts := []byte(str)
+		encBytes := crypto.Keccak256(byts)[:4]
+		for _, arg := range arguments {
+			if !testMode {
+				fmt.Fprintf(os.Stderr, "Scanning: %s\r", input)
+			}
+			if bytes.Equal(encBytes, arg) {
+				prettyPrint(hex.EncodeToString(encBytes), input)
+				// TODO: see issue #1873
+				// wg.Done()
```

Comment:
Regarding issue #1873, this `Done()` is doubled,  because `defer wg.Done()` from line 55 will be always called before this function exits, so maybe this is why you were getting `negative WaitGroup counter`

Reply:
That fixed it. Thanks!